docs/application/native/guides/text-input/multi-assistant.md

```diff
@@ -0,0 +1,444 @@
+  }
+
+  /* Unset */
+  void
+  unset_service_state_changed_cb()
+  {
+      int ret;
+      ret = ma_unset_service_state_changed_cb();
+      if (MA_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Set the active state change callback to be invoked when the active state of current voice assistant changes:
+
+  ```c
+  /* Callback */
+  void
+  __active_state_changed_cb(ma_active_state_e previous, ma_active_state_e current, void* user_data)
+  {
+      /* Your code */
+  }
+
+  /* Set */
+  void
+  set_active_state_changed_cb()
+  {
+      int ret;
+      ret = ma_set_active_state_changed_cb(__active_state_changed_cb, NULL);
+      if (MA_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Unset */
+  void
+  unset_active_state_changed_cb()
+  {
+      int ret;
+      ret = ma_unset_active_state_changed_cb();
+      if (MA_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Set the audio streaming callback to be invoked when the Multi-assistant framework sends user utterance audio data:
+
+  ```c
+  /* Callback */
+  void
+  __audio_streaming_cb(ma_audio_streaming_event_e event, char* buffer, int len, void* user_data);
+  {
+      /* Your code */
+  }
+
+  /* Set */
+  void
+  set_audio_streaming_cb()
+  {
+      int ret;
+      ret = ma_set_audio_streaming_cb(__audio_streaming_cb, NULL);
+      if (MA_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Unset */
+  void
+  unset_audio_streaming_cb()
+  {
+      int ret;
+      ret = ma_unset_audio_streaming_cb();
+      if (MA_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Set the error callback to be invoked when an error occurs while processing the Multi-assistant related tasks:
+
+  ```c
+  /* Callback */
+  void
+  __error_cb(ma_error_e reason, void* user_data)
+  {
+      /* Your code */
+  }
+
+  /* Set */
+  void
+  set_error_cb(ma_h vc)
+  {
+      int ret;
+      ret = ma_set_error_cb(__error_cb, NULL);
+      if (MA_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Unset */
+  void
+  unset_error_cb(ma_h vc)
+  {
+      int ret;
+      ret = ma_unset_error_cb();
+      if (MA_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+<a name="info"></a>
+## Retrieve Multi-assistant information
+
+To get information about the current client state and current language:
+
+- Get the current client state using `ma_get_state()`.
+
+  The client state changes according to function calls when the Multi-assistant client library is, for example, initialized and prepared:
+
+  ```c
+  void
+  get_state()
+  {
+      ma_state_e current_state;
+      int ret;
+      ret = ma_get_state(&current_state);
+      if (MA_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Get the current language with `ma_get_current_language()`. The wake word detection works for the current (default) language. Use the language change callback to get language change notifications.
```

Comment:
Please use colon (:) instead of a period at the end of the sentence as it is followed by code.

- Get the current language with `ma_get_current_language()`. The wake word detection works for the current (default) language. Use the language change callback to get language change notifications:

Reply:
Suggested changes uploaded : 2c1a24a1e02be6d3d3b9fd1b0a1fcfdae4c07534